openedx/taxonomy-connector
# feat: refresh_xblock_skills management command

Adds command `refresh_xblock_skills` to update skills related to passed xblocks, or all xblocks in passed courses or all xblocks in all courses available from `get_course_metadata_provider`.

**Reviewers:**
- [x] @kaustavb12
- [x] @sameenfatima78 

**Test instructions:**

If you try out below commands in project directory, it should fail at ESME/lightcast API call due to incorrect URL and not before that. It should use test settings by default and use mock data for content. If not run `make test` once.

```bash
python manage.py migrate
python manage.py refresh_xblock_skills --xblock block-v1:edX+DemoX+Demo_Course+type@video+block@0b9e39477cf34507a7a48f74be381fdd
python manage.py refresh_xblock_skills --course some-id
python manage.py refresh_xblock_skills --all
```

To test `args-from-database` argument open django shell and run below commands:
```python
from taxonomy.models import RefreshXBlockSkillsConfig
config = RefreshXBlockSkillsConfig.get_solo()
config.arguments = ' --all --commit '
config.save()
```
Then run below command:
`python manage.py refresh_xblock_skills --args-from-database`

*Although, running above commands should cover the code in this MR, for end to end test with real metadata provider from course-discovery, I'll try to come up with test instructions once I figure out way to mock lightcast API or get free access to lightcast API*

**Merge checklist:**
    - `make upgrade && make requirements` have been run to regenerate requirements
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
- [x] [Version](https://github.com/edx/taxonomy-connector/blob/master/taxonomy/__init__.py) bumped
- [x] [Changelog](https://github.com/edx/taxonomy-connector/blob/master/CHANGELOG.rst) record added
- [ ] https://github.com/openedx/openedx-events/pull/143
- [ ] Update openedx-events dependency

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/taxonomy-connector/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [GitHub Actions](https://github.com/edx/taxonomy-connector/actions), verify version has been pushed to [PyPI](https://pypi.org/project/taxonomy-connector/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [course-discovery](https://github.com/edx/course-discovery) to upgrade dependencies (including taxonomy-connector)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in course-discovery will look for the latest version in PyPi.
